server/src/main/webapp/WEB-INF/rails/webpack/views/pages/roles.tsx

```diff
@@ -127,7 +128,7 @@ export class RolesPage extends Page<null, State> {
   }
 
   fetchData(vnode: m.Vnode<null, State>): Promise<any> {
-    return Promise.all([PluginInfoCRUD.all({type: ExtensionTypeString.AUTHORIZATION}), AuthConfigsCRUD.all(), RolesCRUD.all(), EnvironmentCRUD.all()])
+    return Promise.all([PluginInfoCRUD.all({type: ExtensionTypeString.AUTHORIZATION}), AuthConfigsCRUD.all(), RolesCRUD.all(), EnvironmentCRUD.all(), ConfigReposCRUD.all()])
```

Comment:
😕 Going forward, this section of code would make a bunch of API calls based on what entity types are supported.
We should consider this area for a possible piece of improvement going forward to avoid making tons of API calls.

Reply:
How about trying out:

- Sending entity information as part of the embedded `meta`  field on the page (as this information is fetched only on the page load). This approach would avoid all API calls.
- Replacing all the API calls with only one internal API which returns all the entity information.

We can see what works the best for us and then take a decision about the approach!!

Reply:
Yes agreed. We won't address this as a part of this PR. Can you create an issue for the same and link it here so that we can keep track and not lose the context. 
WDYT @GaneshSPatil

Reply:
Make Sense. Created an issue: https://github.com/gocd/gocd/issues/7381